### PR TITLE
Use `class_name` instead of `name` for model_type

### PR DESCRIPTION
**Pull request by arunagw**
We have a case in which we are using [activity gem](https://github.com/chaps-io/public_activity) to capture events. And this is polymorphic too. In many case like this gem few times namespace models with `::` like `PublicActivity::Activity` or something like this. 

As name can consist `::` and it fails with GraphQL [introspection query](https://github.com/graphql/graphql-js/blob/master/src/utilities/introspectionQuery.js) on a regexp parsing.

One way to fix this by using `class_name` which will return `Activity` and remove modules from name.

I am not sure this is the best way to handle/fix this but I am looking for a fix that can work.

### CODE REVIEW

File: lib/graphql/models/definition_helpers/associations.rb [hunk 1]
```diff
@@ -41,7 +41,7 @@ def self.resolve_has_one_type(reflection)
           graph_types = valid_types.map { |t| GraphQL::Models.get_graphql_type(t) }.compact
 
           GraphQL::UnionType.define do
-            name "#{model_type.name}#{reflection.foreign_type.classify}"
+            name "#{model_type.class_name}#{reflection.foreign_type.classify}"
```
Comment (theorygeek) [inline lib/graphql/models/definition_helpers/associations.rb]: Hmm, I'm not sure if this code works universally. At least on Rails 4, ActiveRecord classes don't respond to `class_name`:
```ruby
class MyModel < ActiveRecord::Base; end
MyModel.class_name
# NoMethodError: undefined method `class_name' for MyModel(Table doesn't exist):Class
# Did you mean?  class_eval
```

Comment (theorygeek) [inline lib/graphql/models/definition_helpers/associations.rb]: Perhaps instead, this code could be modified to do some sanitization on the name, stripping it down to only alphanumeric characters:
```ruby
model_type.name.gsub(/[^A-Za-z0-9_]+/, '')
```

Comment (gustavovnicius) [inline lib/graphql/models/definition_helpers/associations.rb]: @theorygeek @arunagw pls check #42

**Comment by theorygeek** [comment]
Changes are incorporated from https://github.com/goco-inc/graphql-activerecord/pull/42

**Comment by arunagw** [comment]
👍

**Pull request closed without merging**